### ISSUE TITLE
write_with_newline should not warn if the string contains multiple newlines

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang-nursery/rust-clippy#3126]
The write_with_newline lint should not fire on a string containing multiple newlines:

```rust
print!("multiple\nlines\nhere\n");
```

Eliminating the last newline and switching to the `ln` version of the macro makes such code *less* clear, if you want to print a set of lines that each end in a newline.

**Comment by oli-obk** [comment]
We're already ignoring strings with `\n\n` at their end. Changing the algorithm to check for any `\n` other than the last should be easy
